libs/awscdk/src/function.ts

```diff
@@ -214,4 +214,25 @@ export class Function
   public get functionName(): string {
     return this.function.functionName;
   }
+
+  /**
+   * @internal
+   */
+  protected _getCodeLines(handler: cloud.IFunctionHandler): string[] {
+    const inflightClient = handler._toInflight();
+    const lines = new Array<string>();
+    const client = "$handler";
+
+    lines.push('"use strict";');
+    lines.push(`var ${client} = undefined;`);
+    lines.push("exports.handler = async function(event, context) {");
+    lines.push("  globalThis.$awsLambdaContext = context;");
```

Comment:
I am concerned that multiple concurrent invocations will corrupt this global state.

Can we use https://nodejs.org/api/async_context.html?

Reply:
@eladb can you explain? Concurrent invocations would have to be running on separate AWS Lambda instances, wouldn't they?

Reply:
Yes, that might be guaranteed by the runtime but not enforced by the programming model. There is a reason why `ctx` is an argument to the handler and not a global.

I would say that the "correct" solution is to use the local storage, and perhaps that's something we want to formalize in wing as well.

but if it's too complex, I would at least add some defensive coding which nullifies the global at the end of the run (in a finally block) and verified that it is null at the beginning.

Reply:
IMO if AWS is providing this as a contract, then it should be alright to optimize our private implementations around it -- the only issue would be if we introduce a bug where we start using these AWS Lambda wrappers on platforms that aren't AWS Lambda.

I like the idea of adding some error checking so at least we can catch if this assumption is broken 👍